server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -7484,7 +7484,13 @@ public void testIndexCommitsListener() throws Exception {
 
         final Engine.IndexCommitListener indexCommitListener = new Engine.IndexCommitListener() {
             @Override
-            public void onNewCommit(ShardId shardId, long primaryTerm, Engine.IndexCommitRef indexCommitRef, Set<String> additionalFiles) {
+            public void onNewCommit(
+                ShardId shardId,
+                Store store,
+                long primaryTerm,
+                Engine.IndexCommitRef indexCommitRef,
+                Set<String> additionalFiles
+            ) {
```

Comment:
Could we assert something about the received store here? For instance:

```suggestion
            public void onNewCommit(
                ShardId shardId,
                Store store,
                long primaryTerm,
                Engine.IndexCommitRef indexCommitRef,
                Set<String> additionalFiles
            ) {
                assertNotNull(store);
                assertTrue(store.hasReferences());
```